src/any_agent/telemetry/base.py

```diff
@@ -130,3 +138,28 @@ def parse_generic_key_value_string(text: str) -> dict[str, str]:
             result[key] = value
 
         return result
+
+    def _extract_telemetry_data(self, telemetry: list[dict[str, Any]]) -> list[dict]:
+        """Extract LLM calls and tool calls from LangChain telemetry."""
+        calls = []
+
+        for span in telemetry:
+            calls.append(self.extract_interaction(span)[1])
+
+        return calls
+
+    def extract_interaction(self, span: dict[str, Any]) -> tuple[str, dict[str, Any]]:
+        """Extract interaction details from a span."""
+        attributes = span.get("attributes", {})
+        span_kind = attributes.get("openinference.span.kind", "")
+
+        if span_kind == "LLM" or "LiteLLMModel.__call__" in span.get("name", ""):
```

Comment:
Could we use an Enum for the `span_kind`?

Reply:
> Could we use an Enum for the span_kind?

Yes, I think it is a good idea. We will be revisiting the telemtetry code soon so will probably add something like that